### PR TITLE
feat: add avatar-size prop to contact info

### PR DESCRIPTION
**Pull request by tonydialpad**
# PR Add `avatar-size` prop to `DtRecipeContactInfo`

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [ ] Refactoring
- [ ] Documentation

## :bulb: Context
As a requirement from this [DP-66474](https://dialpad.atlassian.net/browse/DP-66474), we need to update the size of avatar to size 'md'. The original callbar is using 'lg' as default, and the manage participants modal wants it to 'md', so the best solution is to make the avatar size passed from parent.

## :pencil: Checklist

- [x] I have reviewed my changes
- [ ] I have added tests
- [x] I have added all relevant documentation
- [x] I have validated components with a screen reader
- [x] I have validated components keyboard navigation
- [x] I have considered the performance impact of my change
- [x] I have checked that my change did not significantly increase bundle size
- [ ] I am exporting any new components or constants in the index.js in the component directory
- [ ] I am exporting any new components or constants in the index.js in the root

## :crystal_ball: Next Steps

Make a new release for dialtone-vue

## :camera: Screenshots / GIFs
Size LG:
Screenshot 2023-03-14 at 1 34 59 PM

Size MD:
Screenshot 2023-03-14 at 1 35 08 PM

## :link: Sources

JIRA: https://dialpad.atlassian.net/browse/DP-66474
Vue3 PR: https://github.com/dialpad/dialtone-vue/pull/810

[DP-66474]: https://dialpad.atlassian.net/browse/DP-66474?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by juliodialpad — COMMENTED** [review]
From the technical side this is ok, but let's ask @francisrupert to check that this is ok from the design side.

Maybe we want to limit the recipe to 'md' and 'lg' or is it ok to have all the avatar sizes?

**Comment by braddialpad** [comment]
This should probably be updated to use the new avatar right? This is pretty outdated now

File: recipes/list_items/contact_info/contact_info.vue [hunk 1]
```diff
@@ -131,6 +132,16 @@ export default {
...
 
+    /**
+     * The size of the avatar
+     * @values xs, sm, md, lg, xl
+     */
+    avatarSize: {
+      type: String,
+      default: 'lg',
+      validator: (size) => Object.keys(AVATAR_SIZE_MODIFIERS).includes(size),
+    },
```
Comment (iropolo) [inline recipes/list_items/contact_info/contact_info.vue]: DtAvatar already has validator, so its throwing double console for the same error

I dont know if it is necessary to have this validator also here in the recipe...

image

Comment (tonydialpad) [reply recipes/list_items/contact_info/contact_info.vue]: The validator is removed, please review.

**Comment by tonydialpad** [comment]
Hi @braddialpad , the user status is replaced by avatar's presence, please review again, thanks!

File: recipes/list_items/contact_info/contact_info.stories.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import DtRecipeContactInfo from './contact_info';
 import DtRecipeContactInfoMdx from './contact_info.mdx';
 import DtRecipeContactInfoDefaultTemplate from './contact_info_default.story';
 import DtRecipeContactInfoVariantsTemplate from './contact_info_variants.story';
-import { USER_STATUS_COLOR_MODIFIERS } from './contact_info_constants';
+import { AVATAR_SIZE_MODIFIERS, PRESENCE_STATES_LIST } from '@';
```
Comment (braddialpad) [inline recipes/list_items/contact_info/contact_info.stories.js]: don't use '@' we had some strange cyclical dependency issues related to this. use `@/components/avatar`

File: recipes/list_items/contact_info/contact_info.vue [hunk 3]
```diff
@@ -12,7 +12,8 @@
       >
         <dt-avatar
```
Comment (braddialpad) [inline recipes/list_items/contact_info/contact_info.vue:13]: I'm a bit confused it looks like it's still showing the old presence style? Shouldn't it be a circle?

Screenshot 2023-03-14 at 1 52 35 PM

Comment (tonydialpad) [reply recipes/list_items/contact_info/contact_info.vue:13]: It shows circle in my test:
Screenshot 2023-03-14 at 1 34 59 PM
I guess you may need to do some cache clean up.

Comment (braddialpad) [reply recipes/list_items/contact_info/contact_info.vue:13]: Yeah for some reason the deploy previews have been extremely slow to update lately... it looks fine now.

**Review by braddialpad — APPROVED** [review]

**Pull request merged**